contracts/WorkspaceRegistry.sol

```diff
@@ -170,18 +179,59 @@ contract WorkspaceRegistry is
         bool[] memory _enabled,
         string[] memory _emails
     ) external whenNotPaused onlyWorkspaceAdmin(_id) withinLimit(_members.length) {
-        require(_members.length == _roles.length, "UpdateWorkspaceMembers: Parameters length mismatch");
-        require(_members.length == _enabled.length, "UpdateWorkspaceMembers: Parameters length mismatch");
-        require(_members.length == _emails.length, "UpdateWorkspaceMembers: Parameters length mismatch");
-        for (uint256 i = 0; i < _members.length; i++) {
-            address member = _members[i];
-            /// @notice The role 0 denotes an admin role
-            /// @notice The role 1 denotes a reviewer role
-            uint8 role = _roles[i];
-            bool enabled = _enabled[i];
-            _setRole(_id, member, role, enabled);
-        }
-        emit WorkspaceMembersUpdated(_id, _members, _roles, _enabled, _emails, block.timestamp);
+        _updateWorkspaceMembers(_id, _members, _roles, _enabled, _emails);
+    }
+
+    /**
+     * @notice Create an invite link for someone to join with
+     * @param _id ID of workspace to create invite link for
+     * @param _role what the role of the invited member should be
+     * @param publicKeyAddress Generated public key address for the invite link
+     * (corresponding private key should be sent to invitee)
+     */
+    function createInviteLink(
+        uint96 _id,
+        uint8 _role,
+        address publicKeyAddress
+    ) external whenNotPaused onlyWorkspaceAdmin(_id) {
+        bytes32 apiFlag = _apiFlagForWorkspaceId(_id, _role);
+        AnonAuthoriser(anonAuthoriserAddress).generateAnonAuthorisation(publicKeyAddress, apiFlag);
+    }
+
+    /**
+     * @notice Join a workspace with an invite link
+     * @param _id ID of workspace to join
+     * @param _email email of the user that is requesting to join (optional, can be empty)
+     * @param _role the role the user was invited for
+     * Remaining params are of the signature to be sent to AnonAuthoriser
+     */
+    function joinViaInviteLink(
+        uint96 _id,
+        string memory _email,
+        uint8 _role,
+        uint8 signatureV,
+        bytes32 signatureR,
+        bytes32 signatureS
+    ) external whenNotPaused {
+        bytes32 apiFlag = _apiFlagForWorkspaceId(_id, _role);
+        AnonAuthoriser(anonAuthoriserAddress).anonAuthorise(
+            address(this),
```

Comment:
shouldn't `address(this)` return the contract's address instead of authoriser's address?

Reply:
Yeah, we need the contract's address -- since it was the authoriser